commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java

```diff
@@ -76,12 +80,18 @@
      */
     private volatile int[] groupsIds;
 
+    /**
+     * Some SFTP-only servers disable the exec channel. When exec is disabled, things like getUId() will always fail.
+     */
+    private boolean execDisabled;
+
     protected SftpFileSystem(final GenericFileName rootName, final Session session,
             final FileSystemOptions fileSystemOptions) {
         super(rootName, null, fileSystemOptions);
         this.session = Objects.requireNonNull(session, "session");
         this.connectTimeoutMillis = SftpFileSystemConfigBuilder.getInstance()
                 .getConnectTimeoutMillis(fileSystemOptions);
+        detectExecDisabled();

```

Comment:
@apearc03 

Thank you for updating the ASL headers.

Unless you expect the state of a server to change and re-detect `execDisabled`, please make `execDisabled` `final` and make `detectExecDisabled()` return a `boolean`.  A boolean method would be cleaner anyway IMO. Then the ctor can say 'this.execDisable = detectExecDisabled();`

We can always drop the `final` on the ivar later if `detectExecDisabled()` needs to be called post construction.

Reply:
Made changes to cleanup exec ```execDisabled``` initialization.